demo/src/screens/MainScreen.js

```diff
@@ -174,17 +237,74 @@ class MainScreen extends Component {
         hideUnderline
         floatingPlaceholder={false}
         text70
-        trailingAccessory={
+        leadingAccessory={
           filterText ? (
-            <Button link iconSource={Assets.icons.demo.close} $iconDefault onPress={this.clearSearch}/>
+            <Button link marginR-5 iconSource={Assets.icons.demo.close} $iconDefault onPress={this.clearSearch}/>
           ) : (
-            <Icon tintColor={Colors.$iconDefault} source={Assets.icons.demo.search}/>
+            <View>
+              <Icon tintColor={Colors.$iconDefault} source={Assets.icons.demo.search}/>
+            </View>
           )
         }
       />
     );
   };
 
+  onPress = ({customValue: index}) => {
+    const {chipsLabels} = this.state;
+    this.hasPressItem = true;
+    this.hasUserScrolled = false;
+    this.setState({
+      selectedSection: index,
+      faderStart: index !== 0,
+      faderEnd: index !== chipsLabels.length - 1
+    });
+  };
+
+  onCheckViewableItems = ({viewableItems}) => {
+    const {chipsLabels, selectedSection} = this.state;
+    if (!this.hasPressItem && this.hasUserScrolled) {
+      const title = viewableItems[0].section.title;
+      const sectionIndex = _.findIndex(chipsLabels, c => {
+        return c === title;
+      });
+
+      if (sectionIndex !== -1 && sectionIndex !== selectedSection) {
+        this.setState({
+          selectedSection: sectionIndex,
+          faderStart: sectionIndex !== 0,
+          faderEnd: sectionIndex !== chipsLabels.length - 1
+        });
+      }
+    }
+  };
+
+  renderChip(label, index) {
+    return (
+      <Chip
+        marginH-5
+        marginV-10
+        marginL-15={index === 0}
+        label={label}
+        key={index}
+        containerStyle={index === this.state.selectedSection ? styles.selectedChipContainer : styles.chipContainer}
+        onPress={this.onPress}
+        customValue={index}
+        labelStyle={index === this.state.selectedSection ? styles.selectedChip : undefined}
+      />
+    );
+  }
+
+  renderSectionHeader = ({section}) => {
+    return (
+      <View backgroundColor={'white'}>
+        <Text back marginV-20 marginH-20 text60M>
+          {section.title}
+        </Text>
+      </View>
+    );
+  };
+
   renderItem = ({item}) => {
     const {renderItem} = this.props;
```

Comment:
There's a performance issue in renderItem that worth fixing before we merge this. 
Let's go over it together, ill explain in details...